### PR TITLE
Flesh out documentation, add CLIs for the executables, rig dynamic loading of agent code for players.

### PR DESCRIPTION
**Pull request by jsbohnert**

### CODE REVIEW

**Comment by kyokley** [comment]
I'm definitely a fan of the improved documentation so I'd like to get that in as soon as we can. As for the rest, It'd be nice if we can still allow players to accept args during initialization.

**Comment by jsbohnert** [comment]
Following our discussion, I added a method to pass arguments to players specified via --p1 and --p2.
Instead of forcing a specific format for args (like a config file, or json expectations) what I think works reasonable is providing a way to send in a string that your class is guaranteed to get on init, and let you unpack it any way you want: you can decode it, it could be the name of a config file, w/e.

Example:

main.py --p1 "myagent style=angry, maxthinktime=500"
would pass your class a kwarg called 'argstring' with a value "style=angry, maxthinktime=500"

Basically everything before the first space is your agent name, everything after is your argstring.

Think this is reasonable enough to facilitate testing?

**Comment by kyokley** [comment]
Interesting solution. I'm hesitant to completely change main.py though. Why not just make the command line interface its own thing?

**Comment by jsbohnert** [comment]
Like a new executable that does the same thing? I guess I don't find
main.py on its current form to be user friendly.

The users should only be concerned about editing their ai code, not
maintaining the executable that lets them participate. I'm working from the
assumption that main.py is our recommended tool for them to test their
agent code.

I'm happy to split this into a new testing executable, but what would
main.py be for?

**Comment by kyokley** [comment]
The reason is for people that have already gotten started. I don't want to suddenly break something that is currently working. I'm trying really hard not to make breaking changes since we don't have a way to let everyone know that the system has changed.

**Comment by jsbohnert** [comment]
Ultimately it's fine to split it up, provided the documentation can point
to them both. I'm interested in getting the barrier to entry lowered for
anyone who might take a look, be confused and not participate.

Aside: was there anything in the announcements about where to go with
questions?

**Comment by kyokley** [comment]
Yeah, I'm okay with dropping main.py from the documentation entirely and just describing the CLI going forward.

As for questions, I believe the flyers mention emailing the IT activities committee. However, we're still waiting on approval to put the flyers up.

**Comment by jsbohnert** [comment]
Replaced main.py as discussed. Moved enhanced tooling to battle.py.
